lib/Optimizer/IROptimizer.cpp

```diff
@@ -1246,8 +1366,12 @@ void performPeepholeOptimizations(IRFunction &M) {
 
     // Remove useless copies.
     if (auto *CI = dyn_cast<CopyInst>(I)) {
-      if (getOrigin(CI->getSrc()) == getOrigin(CI->getDest()))
+      auto srcOrigin = getOrigin(CI->getSrc());
+      auto destOrigin = getOrigin(CI->getDest());
+      if ((srcOrigin == destOrigin) &&
+          (srcOrigin->getType()->size() == destOrigin->getType()->size())) {
```

Comment:
If srcOrigin == destOrigin, then `srcOrigin->getType()->size() == destOrigin->getType()->size()` is always true, or? Why do we need the check it? Or may be you meant a different condition?